hw/ip/aes/data/aes.hjson

```diff
@@ -194,11 +209,83 @@
       '''
     }
   ],
+
+  /////////////////////
+  // Countermeasures //
+  /////////////////////
+
   countermeasures: [
     { name: "BUS.INTEGRITY",
       desc: "End-to-end bus integrity scheme."
     }
+    { name: "CONFIG.SHADOW",
+      desc: "Main control register shadowed."
+    }
+    { name: "CONFIG.SPARSE",
+      desc: "Critical fields in main control register one-hot encoded."
+    }
+    { name: "DATA_REG.SW_UNREADABLE",
+      desc: "Key, IV and data input registers are write only for software."
+    }
+    { name: "DATA_REG.SEC_WIPE",
+      desc: "Key, IV, input/output data and internal state registers can be cleared with pseudo-random data."
+    }
+    { name: "MAIN.FSM.SPARSE",
```

Comment:
I'm not sure that this is how the countermeasure instance name is supposed to be used. @msfschaffner: Is this supposed to be allowed? If not, I think there's a missing check in the Python :-)

That said, I think it's a good idea! If it's not currently how we intend things to work, maybe we should explicitly allow it? (In which case, I'd suggest allowing hierarchical instance names, so use `rsplit(..., 2)` in the Python and treat the stuff to the left as an opaque blob)

Reply:
`MAIN` is the optional instance name and `FSM` the asset. This is inline with what @msfschaffner proposed in #9816.

Reply:
So the naming convention we outlined was to allow for an optional uniquifier in front of the canonical name, see also on the spreadsheet where the naming convention has initially been proposed.

I am not sure we should go full-on hierarchical with unbounded number uf hierarchy levels, though.
My concern is that this may just lead to more naming discussions, as it would allow for more naming combinations.

Reply:
@tjaychen

Reply:
yeah the optional prefix was mostly a mechanism to allow the countermeasure writer a way to differentiate among instances of countermeasures.  There are not a lot of strict rules about this.  I would prefer not to go down the path of full hierarchy, because if we do, the question will become how we validate that.